File: examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairCodeThreadCommand.java
package com.matter.controller.commands.pairing;

import chip.devicecontroller.ChipDeviceController;
import com.matter.controller.commands.common.CredentialsIssuer;

public final class PairCodeThreadCommand extends PairingCommand {
  public PairCodeThreadCommand(ChipDeviceController controller, CredentialsIssuer credsIssue) {
    super(controller, "code-thread", PairingModeType.CODE, PairingNetworkType.THREAD, credsIssue);
  }

  @Override
  protected void runCommand() {}
}

File: examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairCodeThreadCommand.kt
/*
 * SPDX-FileCopyrightText: 2023 Project CHIP Authors
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 */
package com.matter.controller.commands.pairing

import chip.devicecontroller.ChipDeviceController
import com.matter.controller.commands.common.CredentialsIssuer

class PairCodeThreadCommand(controller: ChipDeviceController, credsIssue: CredentialsIssuer?) :
  PairingCommand(
    controller,
    "code-thread",
    credsIssue,
    PairingModeType.CODE,
    PairingNetworkType.THREAD
  ) {
  override fun runCommand() {
    currentCommissioner()
      .pairDeviceWithCode(
        getNodeId(),
        getOnboardingPayload(),
        getDiscoverOnce(),
        getUseOnlyOnNetworkDiscovery(),
        null,
        getThreadNetworkCredentials(),
      )
    currentCommissioner().setCompletionListener(this)
    waitCompleteMs(getTimeoutMillis())
  }
}
